### PR TITLE
Refactor private properties

### LINKED ISSUE #1941: Remove _ prefix for private variables
We have a type checker.

### PR DESCRIPTION
**Pull request by domoritz**
Remove _ for privat properties. Use plural for dictionaries of scales, axes, and legends. Make more properties read only. Remove accessor functions for parent, description, padding, children, child, stack, and data.

Fixes #1941

### CODE REVIEW

File: src/compile/unit.ts [hunk 1]
```diff
@@ -59,35 +64,37 @@ export class UnitModel extends Model {
...
     }
 
-    this._axis = this._initAxis(encoding, config);
-    this._legend = this._initLegend(encoding, config);
+    this.axes = this.initAxes(encoding, config);
+    this.legends = this.initLegend(encoding, config);
 
     // width / height
-    this._initSize(mark, this._scale,
+    const {width, height} = this.initSize(mark, this.scales,
```
Comment (kanitw) [inline src/compile/unit.ts]: `{width: this.width, height: this.height}`  more concise?

File: src/compile/unit.ts [hunk 2]
```diff
@@ -320,51 +315,43 @@ export class UnitModel extends Model {
     return UNIT_CHANNELS;
   }
 
-  protected mapping() {
-    return this.encoding();
-  }
-
-  public stack(): StackProperties {
-    return this._stack;
+  protected getMapping() {
```
Comment (kanitw) [inline src/compile/unit.ts]: I don't like unnecessary `get*` prefix for these function.  It makes code longer everywhere.

Comment (domoritz) [inline src/compile/unit.ts]: Okay. I can rename them back. The problem was that I had not renamed scale to scales yet and so the I had a name conflict.

Comment (kanitw) [inline src/compile/unit.ts]: Make sense then.

File: src/compile/unit.ts (hunk 1 above)
Comment (domoritz) [inline src/compile/unit.ts]: This creates unused variables and is arguable less predictable.

Comment (kanitw) [inline src/compile/unit.ts]: Can you explain more?  I don't understand what you mean.

Comment (domoritz) [inline src/compile/unit.ts]: `const {width: this.width, height: this.height}  = ...`

fails linting

**Comment by domoritz** [comment]
Pushed fixes.

**Comment by kanitw** [comment]
cc: @arvind @willium  -- you guys should be aware of this PR by @domoritz, as it will create some conflicts.

**Comment by domoritz** [comment]
It should be as easy as removing `()` from wherever ts tells you that something is broken.

**Comment by willium** [comment]
cool!

**Pull request merged**